app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts

```diff
@@ -172,7 +171,7 @@ export class PipelineConfigProvider {
     if (candidates.length) {
       baseKey = candidates[0].provides;
     }
-    return this.getStageTypes().filter(stageType => {
+    return stageTypes.filter(stageType => {
```

Comment:
`getStageType` returns a deep copy of the stage types. Not sure if that's really important (let's find out?).